crates/red_knot_python_semantic/src/semantic_index.rs

```diff
@@ -109,12 +110,26 @@ pub(crate) fn attribute_assignments<'db, 's>(
     let index = semantic_index(db, file);
     let class_scope_id = class_body_scope.file_scope_id(db);
 
-    ChildrenIter::new(index, class_scope_id).filter_map(|(file_scope_id, maybe_method)| {
-        maybe_method.node().as_function()?;
-        let attribute_table = index.instance_attribute_table(file_scope_id);
+    ChildrenIter::new(index, class_scope_id).filter_map(|(child_scope_id, scope)| {
+        let (function_scope_id, function_scope) =
+            if scope.node().scope_kind() == ScopeKind::Annotation {
+                // This could be a generic method with a type-params scope.
+                // Go one level deeper to find the function scope. There is
+                // just one descendant scope, so we just use `.start`.
```

Comment:
I could add something like the following here, if we think it's useful?
```rs
debug_assert!(descendants.start.as_u32() + 1 == descendants.end.as_u32());
```

Reply:
Seems like it doesn't hurt? No strong feeling either way.

Reply:
Glad we added it:
```
fatal[panic] Panicked at crates/red_knot_python_semantic/src/semantic_index.rs:120:17 when checking `/tmp/mypy_primer/projects/mitmproxy/mitmproxy/tools/web/app.py`: `assertion failed: descendants.start.as_u32() + 1 == descendants.end.as_u32()`
```

Reply:
There is just one *child* scope, but there can be multiple *descendant* scopes (if there are nested functions inside that generic method). Updated the comment and removed the assertion.

Reply:
Should we also add a corpus test or mdtest with a snippet like that, so we don't have to rely on the fuzzer or primer in CI for coverage of that kind of pattern?

Reply:
Yes, thanks. Added.